File: solutions/javascript/line-up/1/line-up.js
export const format = (name, number) => {
  // Validate inputs
  if (!name || typeof name !== 'string') {
    throw new Error('Name must be a non-empty string');
  }
  
  if (!Number.isInteger(number) || number < 1 || number > 999) {
    throw new Error('Number must be an integer between 1 and 999');
  }
  
  // Get the ordinal suffix
  let suffix = 'th'; // default
  
  // Handle special cases: 11, 12, 13
  const lastTwoDigits = number % 100;
  if (lastTwoDigits >= 11 && lastTwoDigits <= 13) {
    suffix = 'th';
  } else {
    // Handle other cases based on last digit
    const lastDigit = number % 10;
    switch (lastDigit) {
      case 1:
        suffix = 'st';
        break;
      case 2:
        suffix = 'nd';
        break;
      case 3:
        suffix = 'rd';
        break;
      default:
        suffix = 'th';
    }
  }
  
  // Construct and return the formatted message
  return `${name}, you are the ${number}${suffix} customer we serve today. Thank you!`;
};

File: solutions/python/line-up/1/line_up.py
"""Generate a customer greeting with an ordinal number."""


def line_up(name: str, number: int) -> str:
    """Create a personalized sentence for the deli customer.

    Args:
        name: The customer's name.
        number: The customer's position in line (1 to 999).

    Returns:
        A string like "Mary, you are the 1st customer we serve today. Thank you!"
    """
    # Determine the correct ordinal suffix
    if 11 <= (number % 100) <= 13:
        suffix = "th"
    else:
        last_digit = number % 10
        if last_digit == 1:
            suffix = "st"
        elif last_digit == 2:
            suffix = "nd"
        elif last_digit == 3:
            suffix = "rd"
        else:
            suffix = "th"

    return f"{name}, you are the {number}{suffix} customer we serve today. Thank you!"